app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,33 @@
+class QueueItemsController < ApplicationController
+  before_filter :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+    queue_video(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.destroy if current_user.queue_items.include?(queue_item)
+    redirect_to my_queue_path
+  end
+
+  private
+
+  def queue_video(video)
+    QueueItem.create(video: video, user: current_user, position: new_queue_item_position) unless current_user_queued_video?(video)
```

Comment:
This line http://cl.ly/image/3r2X0i0d4620 is over 80 characters - to keep lines readable and to make debugging easier please refactor any lines over this limit. You can see this referenced in the style guide here - https://github.com/bbatsov/ruby-style-guide#80-character-limits